File: dotnet-backend/projekatFlutter/Models/Vehicle.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace projekatFlutter.Models
{
    public class Vehicle:IEntity<int>
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(20)]
        public string RegistrationNumber { get; set; } = string.Empty;

        [MaxLength(17)]
        public string? VIN { get; set; }

        [Required]
        [MaxLength(50)]
        public string Make { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        public string Model { get; set; } = string.Empty;

        [Required]
        public int Year { get; set; }

        [Required]
        [MaxLength(50)]
        public string VehicleType { get; set; } = string.Empty;

        [Required]
        [MaxLength(30)]
        public string FuelType { get; set; } = string.Empty;

        [Column(TypeName = "decimal(8,2)")]
        public decimal? FuelTankCapacity { get; set; }

        [Column(TypeName = "decimal(10,2)")]
        public decimal? LoadCapacity { get; set; }

        [Required]
        [MaxLength(30)]
        public string Status { get; set; } = "Active";

        [Required]
        public int CurrentOdometer { get; set; } = 0;

        [MaxLength(450)]
        public string? AssignedDriverId { get; set; }

        public DateTime? PurchaseDate { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation Properties
        [ForeignKey("AssignedDriverId")]
        public User? AssignedDriver { get; set; }

        public ICollection<VehicleLocation> VehicleLocations { get; set; } = new List<VehicleLocation>();
        public ICollection<Trip> Trips { get; set; } = new List<Trip>();
        public ICollection<ObdReading> ObdReadings { get; set; } = new List<ObdReading>();
        public ICollection<MaintenanceRecord> MaintenanceRecords { get; set; } = new List<MaintenanceRecord>();
        public ICollection<FuelRecord> FuelRecords { get; set; } = new List<FuelRecord>();
        public ICollection<DeviceTracker> DeviceTrackers { get; set; } = new List<DeviceTracker>();
        public ICollection<VehicleGeofence> VehicleGeofences { get; set; } = new List<VehicleGeofence>();
    }
}

File: frontend/lib/models/vehicle.dart
class Vehicle {
  final int id;
  final String registrationNumber;
  final String? vin;
  final String make;
  final String model;
  final int year;
  final String vehicleType;
  final String fuelType;
  final double? fuelTankCapacity;
  final double? loadCapacity;
  final String status;
  final int currentOdometer;
  final String? assignedDriverId;
  final DateTime? purchaseDate;
  final DateTime createdAt;
  final DateTime updatedAt;

  Vehicle({
    required this.id,
    required this.registrationNumber,
    this.vin,
    required this.make,
    required this.model,
    required this.year,
    required this.vehicleType,
    required this.fuelType,
    this.fuelTankCapacity,
    this.loadCapacity,
    required this.status,
    required this.currentOdometer,
    this.assignedDriverId,
    this.purchaseDate,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'] as int,
      registrationNumber: json['registrationNumber'] as String,
      vin: json['vin'] as String?,
      make: json['make'] as String,
      model: json['model'] as String,
      year: json['year'] as int,
      vehicleType: json['vehicleType'] as String,
      fuelType: json['fuelType'] as String,
      fuelTankCapacity: json['fuelTankCapacity'] != null
          ? (json['fuelTankCapacity'] as num).toDouble()
          : null,
      loadCapacity: json['loadCapacity'] != null
          ? (json['loadCapacity'] as num).toDouble()
          : null,
      status: json['status'] as String,
      currentOdometer: json['currentOdometer'] as int,
      assignedDriverId: json['assignedDriverId'] as String?,
      purchaseDate: json['purchaseDate'] != null
          ? DateTime.parse(json['purchaseDate'] as String)
          : null,
      createdAt: DateTime.parse(json['createdAt'] as String),
      updatedAt: DateTime.parse(json['updatedAt'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'registrationNumber': registrationNumber,
      'vin': vin,
      'make': make,
      'model': model,
      'year': year,
      'vehicleType': vehicleType,
      'fuelType': fuelType,
      'fuelTankCapacity': fuelTankCapacity,
      'loadCapacity': loadCapacity,
      'status': status,
      'currentOdometer': currentOdometer,
      'assignedDriverId': assignedDriverId,
      'purchaseDate': purchaseDate?.toIso8601String(),
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }
}